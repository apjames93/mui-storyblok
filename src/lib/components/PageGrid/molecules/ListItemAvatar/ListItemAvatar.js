import React from 'react';
import PropTypes from 'prop-types';
import { ListItemAvatar as MuiListItemAvatar } from '@material-ui/core';
import Avatar from '@material-ui/core/Avatar';
import { validComponentsRequired } from 'lib/utils/customProps';
import Storyblok from 'lib/utils/Storyblok';
import { renderComponentsWithBridge } from 'lib/utils/customComponents';
import Icon from 'lib/components/PageGrid/atoms/Icon/Icon';

/**
 * MuiListItemAvatar
 */

export const ListItemAvatar = ({
  content,
  rootClass,
  dataBlokC,
  dataBlokUid,
  storyblokClass,
}) => {
  const styles = Storyblok.arrayToMuiStyles(rootClass);

  return (
    <MuiListItemAvatar
      className={`${styles.root} ${storyblokClass}`}
      data-blok-c={dataBlokC}
      data-blok-uid={dataBlokUid}
    >
      <Avatar>
        {renderComponentsWithBridge({ Icon }, content[0])}
      </Avatar>
    </MuiListItemAvatar>
  );
};

export default ListItemAvatar;

ListItemAvatar.propTypes = {
  /**
   * stroyblok multiselect of css classes
   * Override or extend the styles applied to the component
   * */
  rootClass: PropTypes.arrayOf(PropTypes.string),

  /** MuiIcon Allowed maximum: 1 */
  content(props, propName, componentName) {
    return validComponentsRequired(props, propName, componentName, ['MuiIcon'], 1);
  },

  /** storyblok prop for when in editor to allow click bridge */
  dataBlokC: PropTypes.string,
  /** storyblok prop for when in editor to allow click bridge */
  dataBlokUid: PropTypes.string,
  /** storyblok prop for when in editor to allow click bridge */
  storyblokClass: PropTypes.string,
};

ListItemAvatar.defaultProps = {
  rootClass: [],
  content: [],
  dataBlokC: '',
  dataBlokUid: '',
  storyblokClass: '',
};
