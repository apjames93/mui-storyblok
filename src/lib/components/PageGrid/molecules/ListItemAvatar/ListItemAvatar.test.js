import React from 'react';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import ListItemAvatar from './ListItemAvatar';

function setup() {
  const props = {
    content: [{
      _editable: '<!--#storyblok#{"name": "Icon", "space": "48408", "uid": "947be9f0-47c3-4315-a95a-550f0c560eb5", "id": "307934"}-->',
      component: 'Icon',
      iconName: 'android',
    }],
  };
  const comp = shallow(<ListItemAvatar {...props} />);
  return { comp, props };
}

describe('<ListItemAvatar />', () => {
  it('renders ListItemAvatar', () => {
    const { comp } = setup();
    expect(comp).toBeDefined();
  });

  test('snapshot', () => {
    const { props } = setup();
    const tree = renderer.create(<ListItemAvatar {...props} />);
    expect(tree).toMatchSnapshot();
  });
});
