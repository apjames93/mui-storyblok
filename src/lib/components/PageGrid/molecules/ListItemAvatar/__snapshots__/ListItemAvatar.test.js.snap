// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItemAvatar /> snapshot 1`] = `
<div
  className="MuiListItemAvatar-root makeStyles-root-3"
>
  <div
    className="MuiAvatar-root MuiAvatar-circle MuiAvatar-colorDefault"
  />
</div>
`;
